doc/adr/0008-jetstream-kv.md

```diff
@@ -0,0 +1,326 @@
+# 8. JetStream based Key-Value Stores
+
+|Metadata|Value|
+|--------|-----|
+|Date    |2021-06-30|
+|Author  |@ripienaar|
+|Status  |Partially Implemented|
+|Tags    |jetstream, client, kv|
+
+## Context
+
+This document describes a design and initial implementation of a JetStream backed key-value store. The initial implementation
+is available in the CLI as `nats kv` with an initial reference client package implementation in `jsm.go` repository to support
+the CLI.
+
+This document aims to guide client developers in implementing this feature in the language clients we maintain.
+
+## Overview
+
+We intend to hit a basic initial feature set as below, with some future facing goals as indicated:
+
+Initial feature list:
+
+ * Multiple named buckets full of keys with n historical values kept per value
+ * Put and Get of `string(k)=string(v)` values
```

Comment:
I would make the value opaque - []byte

Reply:
however the language implementation like, if its bytes or string, whatever works.